azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/message_queue.py

```diff
@@ -0,0 +1,129 @@
+from six.moves import queue
+import logging
+from threading import Event
+from azure.iot.hub.devicesdk.transport import constant
+
+logger = logging.getLogger(__name__)
+
+
+class MessageQueue(
+    queue.Queue, object
+):  # add object as multiple inheritance b/c Queue is old style (no inheritance from object)
+    def __init__(self, on_enable, on_disable):
+        super(MessageQueue, self).__init__()
+        self._enabled = False
+        self._on_enable = on_enable
+        self._on_disable = on_disable
+
+    def enable(self):
+        if not self._enabled:
+            self._on_enable()  # must do this before changing status!
+            self._enabled = True
+
+    def disable(self):
+        if self._enabled:
+            self._enabled = False
+            self._on_disable()  # must do this after changing status!
+
+    @property
+    def enabled(self):
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, v):
+        if v:
+            self.enable()
+        else:
+            self.disable()
+
+
+class MessageQueueManager(object):
+    def __init__(self, enable_feature_fn, disable_feature_fn):
+        self._enable_feature_fn = enable_feature_fn
+        self._disable_feature_fn = disable_feature_fn
+        self.c2d_message_queue = MessageQueue(
+            self._c2d_message_queue_on_enable, self._c2d_message_queue_on_disable
+        )  # not yet enabled
+        self.input_message_queues = {}
```

Comment:
>{} [](start = 36, length = 2)

if it is a dictionary may be use `dict()`

Reply:
Python convention is to use literals + there is reduced performance from using constructors.

---
In reply to: [259175712](https://github.com/Azure/azure-iot-sdk-python-preview/pull/49#discussion_r259175712) [](ancestors = 259175712)